maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/derive/schema/utils/ReadFileUtils.java

```diff
@@ -0,0 +1,165 @@
+      listOfMessages.add((ObjectNode) arrayNode.get(i));
+    }
+    return listOfMessages;
+  }
+
+  /**
+   * Given a file with ObjectNodes which are line separated,
+   * list of ObjectNodes are generated and returned.
+   * Eg, Expected File contents are :
+   * <p>
+   * {"name":"J"}
+   * </p>
+   * <p>
+   * {"name":"K"}
+   * </p>
+   * <p>
+   * {"name":"L"}
+   * </p>
+   * </p>
+   *
+   * @param content Name of the file to read
+   * @return List of ObjectNodes
+   */
+  public static List<Object> readLinesOfMessages(String content) throws JsonProcessingException {
+
+    List<Object> listOfMessages = new ArrayList<>();
+    String[] lineSeparatedMessages = content.split("\n");
+    for (String line : lineSeparatedMessages) {
+      listOfMessages.add(mapper.readValue(line, ObjectNode.class));
+    }
+
+    return listOfMessages;
+  }
+
+  private static void checkEmpty(String content) {
+    if (content == null || content.length() == 0) {
+      throw new IllegalArgumentException("Input file is empty.");
+    }
+  }
+
+  private static List<Object> readCustom(String content) {
+
+    checkEmpty(content);
+    List<Object> messageList = new ArrayList<>();
+
+    try {
+      messageList.addAll(readArrayOfMessages(content));
+      logger.info("Read input as array of Messages.");
+      return messageList;
+    } catch (Exception ignored) {
+      logger.warn("Cannot be read input as array of Messages.");
+    }
+
+    try {
+      messageList.addAll(readLinesOfMessages(content));
+      logger.info("Read input as lines of messages.");
+      return messageList;
+    } catch (Exception ignored) {
+      logger.warn("Cannot be read input as lines of messages.");
+    }
+
+    try {
+      messageList.add(mapper.readValue(content, ObjectNode.class));
+      logger.info("Read input as ObjectNode.");
+      return messageList;
+    } catch (Exception ignored) {
+      logger.warn("Cannot be read input as ObjectNode.");
+    }
+
+    if (messageList.isEmpty()) {
+      logger.error("Unable to read messages.");
+      throw new IllegalArgumentException("Input file format not understood.");
+    }
+
+    return messageList;
+  }
+
+  public static List<String> readMessagesToString(String content) {
+
+    logger.info(String.format("Reading input, Substring 0-20 is :%n  %s",
+        content.substring(0, min(20, content.length()))));
+
+    List<Object> listOfMessages = readCustom(content);
+    List<String> listOfStrings = new ArrayList<>();
+    for (Object message : listOfMessages) {
+      listOfStrings.add(message.toString());
+    }
+    return listOfStrings;
+  }
+
+  public static List<String> readMessagesToString(File file) throws IOException {
+
+    if (file == null) {
+      throw new NullPointerException("Input file not set.");
+    }
+    logger.info(String.format("Reading input file %s", file.getName()));
+    String fileContent = readFile(file.getAbsolutePath());
+    return readMessagesToString(fileContent);
+  }
```

Comment:
Can we not directly use Arrays.asList(objectMapper.readValue( new File(file),..) rather than converting to string and then to ArrayList

Reply:
The input string could be in any format: array of messages, line separated message or just one message. I read the file and then try which format works.